### ISSUE TITLE
[REQ] Context manager similar to torch.no_grad

### ISSUE THREAD

**Issue by JonathanKuelz** [NVIDIA/warp#590]
### Description
It would be great if warp would offer a feature similar to PyTorch's `with torch.no_grad()` context.

### Context
There are various use cases for such a feature. An obvious one would be efficiency in case only certain operations within a tape are relevant. Another one could be implementing a custom integrator that remains differentiable, but excludes often noisy contact computations from the scope of a tape. With this feature, it would be as easy as inheriting from an existing simulator and putting contact force computations within such a context.

If I am not mistaken, there are currently two different ways of achieving something similar:

1. Implementing a custom adjoint kernel that prevents gradient computation.
2. Manually interrupting the computation graph by passing detached copies of arrays to the kernels that should not contribute to gradient compuation.

Both are suboptimal both in terms of code readability and/or efficiency.
